src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7190,6 +7181,12 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_ImplicitlyTypedDefaultParameter" xml:space="preserve">
     <value>Implicitly typed lambda parameter '{0}' cannot have a default value.</value>
   </data>
+  <data name="ERR_OptionalParamValueMismatch" xml:space="preserve">
+    <value>Parameter {0} has default value which does not match corresponding parameter in delegate type '{1}'.</value>
```

Comment:
Feels like this should be `does match _the_ corresponding parameter`. We should confirm what this looks like when the delegate type is itself an anonymous lambda type: I suspect it won't be readable. I think we should probably use the default values themselves as the arguments.

Reply:
Ah yes there should be an extra `_the_`. I did initially try to include the default parameter values in the error message, but this is a problem if there is an error with one of the default values (i.e., if it is non constant then it is simply printed as `bad`) which is not very readable. 

That being said, we can perhaps suppress this diagnostic if one of the default values failed to bind, and just allow that error to be reported so we aren't trying to print out an invalid constant value.